packages/block-editor/src/components/inserter/index.js

```diff
@@ -183,14 +183,11 @@ export default compose( [
 			getBlockRootClientId,
 			hasInserterItems,
 			__experimentalGetAllowedBlocks,
-			getBlockSelectionEnd,
 		} = select( 'core/block-editor' );
 		const { getBlockVariations } = select( 'core/blocks' );
 
 		rootClientId =
-			rootClientId ||
-			getBlockRootClientId( clientId || getBlockSelectionEnd() ) ||
```

Comment:
The issue was that `rootClientId` and `clientId` are `undefined` for the root appender, so this fallback to `getBlockSelectionEnd` would result in blocks being inserted in the wrong place (and potentially the wrong inserter items displayed).